ocaml/xapi/repository_helpers.ml

```diff
@@ -231,6 +231,22 @@ let assert_gpgkey_path_is_valid path =
     raise Api_errors.(Server_error (invalid_gpgkey_path, [path]))
   )
 
+let assert_remote_pool_url_is_valid ~url =
+  try
+    let uri = Uri.of_string url in
+    match (Uri.scheme uri, Uri.host uri, Uri.path uri) with
+    | Some "https", Some host, path
+      when path = Constants.get_enabled_repository_uri ->
+        Helpers.assert_is_valid_ip `ipv4or6 "url" host
+    | _ ->
+        raise Api_errors.(Server_error (invalid_base_url, [url]))
+  with
```

Comment:
AFAICT `Uri.of_string` does not ever throw an exception, and the only function that will error out is `assert_is_valid_ip`? If that is the case, can we combine these two `raise` together? Something like
```
 let uri = Uri.of_string url in
    match (Uri.scheme uri, Uri.host uri, Uri.path uri) with
    | Some "https", Some host, path
      when path = Constants.get_enabled_repository_uri && Helpers.is_valid_ip `ipv4or6 host->
        ()
    | _ ->
        raise Api_errors.(Server_error (invalid_base_url, [url]))
```